From bf8da26adf08db15ae2cbaeadb40f62af6c52037 Mon Sep 17 00:00:00 2001
From: Noriko Hosoi <nhosoi@redhat.com>
Date: Wed, 29 Jul 2015 11:26:22 -0700
Subject: [PATCH 31/39] Ticket #48232 - winsync lastlogon attribute not syncing
 between DS and AD.

Bug Description:
From Microsoft forum:
 The DirSync control taps into the replication stream to get the necessary
 changes. Since lastLogon is not replicated, it isn't available via the
 DirSync control.
Additional notes:
 The lastLogon attribute is not replicated.
 In contrast the lastLogontimeStamp attribute is replicated.

Fix Description:
 Instead of lastLogon|lastLogoff, sync lastLogonTimestamp|lastLogoffTimestamp
 which are the target of DirSync, to ntUserLastLogon|ntUserLastLogoff.

https://fedorahosted.org/389/ticket/48232

Reviewed by rmeggins@redhat.com (Thank you, Rich!!)

(cherry picked from commit 0db29788e6c1b17f944fcafa368b66580e1e90d5)
(cherry picked from commit b81adb0bc8ad97fec50fba30454e94858476bad5)
---
 ldap/servers/plugins/replication/windows_protocol_util.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ldap/servers/plugins/replication/windows_protocol_util.c b/ldap/servers/plugins/replication/windows_protocol_util.c
index 4cfa20d..5c12af7 100644
--- a/ldap/servers/plugins/replication/windows_protocol_util.c
+++ b/ldap/servers/plugins/replication/windows_protocol_util.c
@@ -194,8 +194,8 @@ static windows_attribute_map user_attribute_map[] =
 {
 	{ "homeDirectory", "ntUserHomeDir", bidirectional, always, normal},
 	{ "scriptPath", "ntUserScriptPath", bidirectional, always, normal},
-	{ "lastLogon", "ntUserLastLogon", fromwindowsonly, always, normal},
-	{ "lastLogoff", "ntUserLastLogoff", fromwindowsonly, always, normal},
+	{ "lastLogonTimestamp", "ntUserLastLogon", fromwindowsonly, always, normal},
+	{ "lastLogoffTimestamp", "ntUserLastLogoff", fromwindowsonly, always, normal},
 	{ "accountExpires", "ntUserAcctExpires", bidirectional, always, normal},
 	{ "codePage", "ntUserCodePage", bidirectional, always, normal},
 	{ "logonHours", "ntUserLogonHours", bidirectional, always, normal},
-- 
1.9.3

